wallet/src/account/output_cache/mod.rs

```diff
@@ -784,16 +818,22 @@ impl OutputCache {
     }
 
     pub fn add_tx(&mut self, tx_id: OutPointSourceId, tx: WalletTx) -> WalletResult<()> {
-        let already_present = self.txs.get(&tx_id).is_some_and(|tx| !tx.state().is_abandoned());
+        let already_present = self.txs.get(&tx_id).is_some_and(|tx| match tx.state() {
+            TxState::Abandoned | TxState::Conflicted(_) => false,
+            TxState::Confirmed(_, _, _) | TxState::InMempool(_) | TxState::Inactive(_) => true,
+        });
         let is_unconfirmed = match tx.state() {
             TxState::Inactive(_)
             | TxState::InMempool(_)
             | TxState::Conflicted(_)
             | TxState::Abandoned => true,
             TxState::Confirmed(_, _, _) => false,
         };
+
         if is_unconfirmed && !already_present {
             self.unconfirmed_descendants.insert(tx_id.clone(), BTreeSet::new());
+        } else {
```

Comment:
this should be `else if !is_unconfirmed`, we don't want to remove it if it is unconfirmed and already present.